cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -286,17 +299,29 @@ public String cloneRemote()
     }
     // Clean up the directory if it already exists.
     deletePathIfExists(getRepositoryRoot());
-    RepositoryConfig repositoryConfig = sourceControlConfig
-        .getRepositoryConfig();
-    CloneCommand command =
-        createCommand(Git::cloneRepository).setURI(repositoryConfig.getLink())
-            .setDirectory(getRepositoryRoot().toFile());
+    RepositoryConfig repositoryConfig = sourceControlConfig.getRepositoryConfig();
+    CloneCommand command = createCommand(Git::cloneRepository).setURI(
+        repositoryConfig.getLink()).setDirectory(getRepositoryRoot().toFile());
     String branch = getBranchRefName(repositoryConfig.getDefaultBranch());
     if (branch != null) {
       command.setBranchesToClone(Collections.singleton((branch)))
           .setBranch(branch);
     }
+    long cloneStartTimeMillis = System.currentTimeMillis();
     git = command.call();
+    long cloneTimeMillis = System.currentTimeMillis() - cloneStartTimeMillis;
+    // Record the repository size metric.
+    try {
+      // Prevent division by zero.
+      long repoSize = Math.max(calculateDirectorySize(getRepositoryRoot()), 1);
+      metricsContext.gauge(SourceControlManagement.METRIC_CLONE_LATENCY,
+          cloneTimeMillis * MEGA_BYTES / repoSize);
```

Comment:
How are we planning on using this metric? Why would a rate be more useful than the total clone time?

Reply:
Clone time is highly dependant on repository size. If we just store clone time it will be difficult for us to correlate with the repo size at that point and derive any useful performance metrics.
Storing clone time per size unit will give us a useful performance indicator and also a measurement to improve in later iterations if we want.

Reply:
I've added a metric for only the clone duration also. 
We could estimate `clone time / repo size` using the metics for `clone time` and `repo size` but that can give imprecise values when the division is done after aggregating across namespaces. So I've kept the separate metric for `clone time / repo size`.